projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js

```diff
@@ -99,33 +109,41 @@ export default function Player( { isSelected, attributes } ) {
 	}, [ preview, isPlayerLoaded, isRequestingEmbedPreview, previewCheckAttempts ] );
 
 	const onSandboxMessage = useCallback( message => {
-		if ( message.event === 'videopress_loading_state' && message.state === 'loaded' ) {
-			setIsPlayerLoaded( true );
+		switch ( message.event ) {
+			case 'videopress_ready':
+				setIsPlayerReady( true );
+				break;
+			case 'videopress_loading_state':
+				setIsPlayerLoaded( message?.state === 'loaded' );
+				break;
 		}
 	}, [] );
 
-	const loadingStyle = {};
-	if ( ! isPreviewReady( preview ) ) {
-		loadingStyle.height = 250;
-	}
-
-	const renderEmbed = () => {
-		if ( html ) {
-			return (
-				<SandBox
-					html={ html }
-					onWindowEvents={ { message: onSandboxMessage } }
-					viewportProps="user-scalable=0"
-				/>
-			);
-		}
-		return <Text>{ __( 'Loading', 'jetpack-videopress-pkg' ) }</Text>;
-	};
+	const loadingOverlay = (
+		<View style={ style[ 'videopress-player__overlay' ] } >
+			<View style={ loadingViewStyle } >
+				<Icon icon={ VideoPressIcon } size={ iconStyle?.size } style={ iconStyle } />
+				<Text style={ style[ 'videopress-player__loading-text' ] }>
+					{ __( 'Loading', 'jetpack-videopress-pkg' ) }
+				</Text>
+			</View>
+		</View>
+	);
 
+	// Show the loading overlay when: 
+	// 1. Player is not ready
+	// 2. Player is loaded but preview is not ready
+	const showLoadingOverlay = ! isPlayerReady || ( isPlayerLoaded && ! isPreviewReady );
```

Comment:
We still need the `isPlayerLoaded` state to show the VP processing window if the preview is not ready.

@fluiddot I noticed in the screen recording with your patch applied the video adjusts to the correct aspect ratio while the video is processing. While that can happen from time to time the more common behavior I see is that the preview is not ready till a few moments after the video is processed. So in these cases we need to hide the loading screen while the preview is not ready but the player has loaded.